agent/xds/endpoints.go

```diff
@@ -246,8 +253,9 @@ func makeLoadAssignment(
 
 			for _, chk := range ep.Checks {
 				if chk.Status == api.HealthCritical {
-					// This can't actually happen now because health always filters critical
-					// but in the future it may not so set this correctly!
+					healthStatus = envoycore.HealthStatus_UNHEALTHY
+				}
+				if endpointGroup.OnlyPassing && chk.Status == api.HealthWarning {
```

Comment:
nit: would be be better to change this to `endpointGroup.OnlyPassing && chk.Status != api.HealthPassing` to match the semantics in case we ever add additional states? For example I think it's possible for checks to have `api.HealthMaint` which should be treated as critical (i.e. don't route traffic) but I don't know if we expose that here so 🤷‍♂.

Reply:
done